neon_mq_connector/connector.py

```diff
@@ -472,10 +503,35 @@ def stop_sync_thread(self):
             self._sync_thread.cancel()
             self._sync_thread = None
 
+    def observe_consumers(self):
+        LOG.debug('Observers state observation')
+        for consumer_name, consumer_data in self.consumers.items():
+            consumer_instance = consumer_data.get('instance')
+            if not (isinstance(consumer_instance, ConsumerThread)
+                    and consumer_instance.is_alive()
+                    and consumer_instance.is_consuming):
+                LOG.info(f'Consumer "{consumer_name}" is dead, restarting')
```

Comment:
Is there a reason this happens we can determine or an exception somewhere to catch?

Reply:
> Is there a reason this happens we can determine or an exception somewhere to catch?

Yes, we have a callback on exception inside the consumer itself (the one you previously implemented), but the problem is that once Thread is joined, it is wiped out from stack so there is no way to restart it, so we need to eliminate it and create a clone of this consumer

Reply:
That makes sense; so, this catches any case where the callback  doesn't handle restarting the consumer.